src/pages/index.tsx

```diff
@@ -65,25 +84,42 @@ const Home: NextPage<DefaultSeoInfo> = (props) => {
 		dispatch(setShowAuthModal(false));
 	};
 
+	/**
+	 * Effects
+	 */
+
+	useEffect(() => {
+		if (isVisitorSpaceNotFoundError && isContentPageNotFoundError) {
+			window.open(`${publicRuntimeConfig.PROXY_URL}/not-found`, '_self');
+		}
+	}, [isVisitorSpaceNotFoundError, isContentPageNotFoundError]);
+
+	useEffect(() => {
+		dispatch(setShowZendesk(true));
+	}, [dispatch]);
+
 	/**
 	 * Render
 	 */
 
 	const renderPageContent = () => {
-		if (!hasCheckedLogin && isBrowser()) {
-			return <Loading fullscreen owner="root index page" />;
+		dispatch(setShowZendesk(true));
+
+		if (isVisitorSpaceLoading || isContentPageLoading) {
+			return <Loading fullscreen owner="slug page: render page content" />;
+		}
+		if (visitorSpaceInfo) {
+			dispatch(setShowZendesk(false));
+			return <VisitorSpaceSearchPage />;
 		}
-		if (isLoggedIn && !!user) {
-			if (showLinkedSpaceAsHomepage && linkedSpaceSlug) {
-				return <Loading fullscreen owner="root page logged" />;
-			}
-			return <LoggedInHome {...props} />;
+		if (contentPageInfo) {
+			return <ContentPageRenderer path={'/' as string} userGroupId={user?.groupId} />;
```

Comment:
You can load this component with path / directly. No need to check for visitorSpaceSearchPage
this DynamicRouteResolver component can be renamed to Home or Homepage